Sources/SwiftSyntax/Documentation.docc/Working with SwiftSyntax.md

```diff
@@ -0,0 +1,221 @@
+
+At the leaves of the syntax tree are ``TokenSyntax`` values. A token syntax 
+value represents an atomic unit of syntax and any associated trivia, like an 
+identifier and the whitespace around it. Taken together, the tokens of a syntax 
+tree represent the lexical structure of the source code. This structure is used
+by linters and formatters to analyze the textual content of source code. 
+
+## Syntax Trivia
+
+Trivia includes whitespace, comments, compiler directives, and erroneous source
+code. Trivia contributes most meaningfully to the lexical structure of a 
+document, but often has little bearing on its syntactic structure. For this
+reason, tools like compilers often discard trivia during source code processing.
+However, maintaining trivia is important for tools like editors, IDEs, 
+formatters, and refactoring engines. SwiftSyntax represents trivia explicitly
+with the ``Trivia`` type. Trivia associated with token syntax can be inspected 
+with  the ``TokenSyntax/leadingTrivia-7p6tp`` and 
+``TokenSyntax/trailingTrivia-5jclz`` accessors.
+
+## Navigating the Syntax Tree
+
+The SwiftSyntax library provides a number of ways of traversing the syntactic
+structure of a syntax tree. Every syntax node includes a reference to its
+parent node via the ``SyntaxProtocol/parent`` accessor, and a reference to its
+child nodes via the ``SyntaxProtocol/children(viewMode:)`` method. The children 
+of a syntax tree always appear in order and the associated `viewMode` allows
+the client to express their intent to process missing and unexpected syntax. 
+
+Syntax nodes also contain strongly-typed APIs to access those same child nodes
+individually. For example, ``ClassDeclSyntax`` provides an 
+``ClassDeclSyntax/identifier`` to get the name of the class, as well as
+``ClassDeclSyntax/members`` to get the syntax node representing the braced block 
+with its members.
+
+Most syntax analyzers want to work with more than one kind of syntax at a time,
+and traverse this syntax in a regular order starting from the root and working
+towards the leaves. SwiftSyntax provides a standard suite of classes and 
+protocols for working with syntax traversals with a visitor-style API. 
+
+A ``SyntaxVisitor`` can be used to walk source code from the root to its leaves.
+To inspect a particular kind of syntax node, a client needs to override the 
+corresponding `visit` method accepting that kind of syntax. The visit methods 
+are invoked as part of an in-order traversal of the syntax tree. To provide a 
+post-order traversal, the corresponding `visitPost` methods can be overridden.
+
+To provide a source transformation, use a ``SyntaxTransformVisitor``.
+Types conforming to this protocol are given the chance to transform the nodes of
+a syntax tree into a user-defined result type. This can be anything from new
+syntax nodes in the case of a formatter, to an integer counter in the case of
+a lexical analyzer counting the number of lines in the source code.
```

Comment:
Do you think we should also mention `SyntaxRewriter` here?

Reply:
Done.